boot/bootutil/src/loader.c

```diff
@@ -1916,3 +1922,189 @@ split_go(int loader_slot, int split_slot, void **entry)
     free(sectors);
     return rc;
 }
+
+#else /* !MCUBOOT_DIRECT_XIP */
```

Comment:
The comment here should be `/* MCUBOOT_DIRECT_XIP */` which describes when the code below will be compiled in (since the above was an ifndef.